src/praveen/narcissistic.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * Function to find that the given number is Narcissistic Number or not
+ * @param {number} num number to check if it is Narcissistic Number
+ * @return {boolean} true or false
+ */
+const checkIfNarcissisticNumber = (num: number): boolean => {
+  if (num < 0) {
+    return false;
+  }
+  const numberStr = num.toString();
+  const numberLength = numberStr.length;
+  let sum = 0;
+  for (let i = 0; i < numberLength; i++) {
+    const digit = parseInt(numberStr.substr(i, 1));
+    sum += Math.pow(digit, numberLength);
+  }
+  return sum === num;
+};
+
+/**
+ * Function to find all the Narcissistic Numbers in the given range
+ * @param {number} startNum starting number in the range
+ * @param {number} endNum ending number in the range
+ * @return {number[]} A array of Narcissistic Numbers
+ */
+const getNarcissisticNumbersWithinRange = (
+  startNum: number,
+  endNum: number
+): Array<number> => {
+  const narcissisticNumbers: Array<number> = [];
+  if (startNum < 0 || endNum < 0 || endNum < startNum) {
```

Comment:
I've a concern with this if condition, for example startNum= -100 and endNum= 10000, we should get output as [153, 370, 371, 407, 1634, 8208, 9474] But with this logic I will get empty array , am I right?

Reply:
fixed